docs/sc-memory/api/cpp/extended/agents/events.md

```diff
@@ -0,0 +1,539 @@
+
+```cpp
+...
+auto const [elementAddr1, elementAddr2] = event.GetConnectorIncidentElements();
+...
+```
+
+### **ScEventEraseOutgoingArc** and **ScEventEraseIncomingArc**
+
+`ScEventEraseOutgoingArc` is class that represents sc-event of erasing outgoing sc-arc from specified sc-element. `ScEventEraseIncomingArc` represents sc-event of erasing incoming sc-arc to specified sc-element.
+
+#### **GetErasableArc**
+
+Method `GetErasableArc` returns erasable sc-arc from (for `ScEventEraseOutgoingArc`) or to (for `ScEventEraseIncomingArc`) listen sc-element (subscription sc-element).
+
+```cpp
+...
+ScAddr const arcAddr = event.GetErasableArc();
+...
+```
+
+#### **GetErasableArcType**
+
+It returns sc-type of erasable sc-arc.
+
+```cpp
+...
+ScAddr const arcType = event.GetErasableArcType();
+...
+```
+
+#### **GetArcSourceElement**
+
+To get source and target sc-elements of erasable sc-arc you can use `GetArcSourceElement` and `GetArcTargetElement` methods. For `ScEventEraseOutgoingArc` method `GetArcSourceElement` returns sc-address of listen sc-element, for `ScEventEraseIncomingArc` method `GetArcTargetElement` returns sc-address of listen sc-element.
+
+```cpp
+...
+ScAddr const arcSourceElement = event.GetArcSourceElement();
+...
+```
+
+#### **GetArcTargetElement**
+
+```cpp
+...
+ScAddr const arcTargetElement = event.GetArcTargetElement();
+...
+```
+
+### **ScEventEraseEdge**
+
+This class represents sc-event of erasing sc-edge from or to specified sc-element.
+
+#### **GetErasableEdge**
+
+Method `GetErasableEdge` returns erasable sc-edge from or to listen sc-element (subscription sc-element).
+
+```cpp
+...
+ScAddr const edgeAddr = event.GetErasableEdge();
+...
+```
+
+#### **GetErasableEdgeType**
+
+It returns sc-type of erasable sc-edge.
+
+```cpp
+...
+ScAddr const edgeType = event.GetErasableEdgeType();
+...
+```
+
+#### **GetEdgeIncidentElements**
+
+To get incident sc-elements of erasable sc-edge you can use `GetEdgeIncidentElements` method.
+
+```cpp
+...
+auto const [elementAddr, targetElementAddr] = event.GetEdgeIncidentElements();
+...
+```
+
+### **ScEventEraseElement**
+
+This class represents sc-event of erasing specified listen sc-element. You can use all methods from objects of this class that are accessible from `ScElementaryEvent` class.
+
+### **ScEventChangeLinkContent**
+
+This class represents sc-event of changing content for listen sc-link. You can use all methods from objects of this class that are accessible from `ScElementaryEvent` class.
+
+--- 
+
+## **Frequently Asked Questions**
+
+<!-- no toc -->
+- [Is there sc-event of adding sc-node?](#is-there-sc-event-of-adding-sc-node)
+- [Is fact of what happened recorded in the knowledge base? Are sc-events recorded in the knowledge base?](#is-fact-of-what-happened-recorded-in-the-knowledge-base-are-sc-events-recorded-in-the-knowledge-base)
+
+### **Is there sc-event of adding sc-node?**
```

Comment:
I propose to append this:

Also, even if events were defined differently, it doesn't negate the fact that one cannot subscribe to sc-event of creating sc-node, since we don't know about that node in advance, because it doesn't exist.